src/java/picard/analysis/CollectAlignmentSummaryMetrics.java

```diff
@@ -116,6 +116,10 @@ public static void main(final String[] argv) {
                     "in the file are aligned then alignment summary metrics collection will fail.");
         }
 
+        if (super.REFERENCE_SEQUENCE != null && this.REFERENCE_SEQUENCE == null) {
```

Comment:
@jacarey CollectAlignmentSummaryMetrics defaults REFERENCE_SEQUENCE to Defaults.REFERENCE_FASTA, but that itself is null unless you supply one with -Dsamjdk.reference_fasta= as far as I can tell.